app/eventPage/[id].tsx

```diff
@@ -66,23 +81,56 @@ export default function EventPage() {
                 (attendee) =>
                   typeof attendee !== "number" && (
                     <SizableText key={attendee.id}>{attendee.name}</SizableText>
-                  ),
+                  )
               )}
             </>
           )}
         </YStack>
       </ScrollView>
-      {(typeof event.organizer !== "number" &&
-        user?.id === event.organizer.id) || (
-        <FloatingActionButton
-          iconAfter={
-            <IconSymbol name="square.and.pencil" size={24} color="$color8" />
-          }
-          // onPress={() => navigation.navigate("eventEdit", { eventId: event.docs[0].id })}
-        >
-          Edit Event
-        </FloatingActionButton>
-      )}
+      {joinStatus !== "joined" &&
+        typeof event.organizer !== "number" &&
+        user?.id === event.organizer.id && (
+          <FloatingActionButton
+            iconAfter={
+              <IconSymbol name="square.and.pencil" size={24} color="$color8" />
+            }
+            // onPress={() => navigation.navigate("eventEdit", { eventId: event.docs[0].id })}
+          >
+            Edit Event
+          </FloatingActionButton>
+        )}
+      {joinStatus !== "joined" &&
+        typeof event.organizer !== "number" &&
+        user?.id !== event.organizer.id && (
+          <FloatingActionButton
+            iconAfter={
```

Comment:
Instead of attaching a listener here, use the `Link` component together with the button as we have done in other places

Reply:
Is there a specific reason for this? or just preference because in my opinion, it looks way more cluttered :/ or am I missing another way of passing the params?

```js    
<Link
push
href={`/joinEvent/attendeeDetails?code=${encodeURIComponent(event.eventCode)}&id=${encodeURIComponent(event.id)}`}
> 
````

Reply:
The main reason is accessibility and semantics. I don't know if the snippet you showed is the only way to go about it, but yeah the Link component should be used whenever possible!